romancal/multiband_catalog/detection_image.py

```diff
@@ -0,0 +1,149 @@
+import logging
+import warnings
+
+import numpy as np
+from astropy.convolution import convolve_fft
+from astropy.utils.exceptions import AstropyUserWarning
+from photutils.segmentation import make_2dgaussian_kernel
+
+from romancal.datamodels import ModelLibrary
+
+log = logging.getLogger(__name__)
+log.setLevel(logging.DEBUG)
+
+
+def make_gaussian_kernel(fwhm):
+    """
+    Make a normalized 2D circular Gaussian kernel.
+
+    The kernel will have odd sizes in both X and Y, be centered in the
+    central pixel, and normalized to sum to 1.
+
+    Parameters
+    ----------
+    fwhm : float
+        The full-width at half-maximum (FWHM) in pixels of the 2D
+        Gaussian kernel.
+
+    Returns
+    -------
+    kernel : `astropy.convolution.Kernel2D`
+        The output Gaussian kernel.
+    """
+    size = np.ceil(3 * fwhm).astype(int)
+    size = size + 1 if size % 2 == 0 else size  # make size be odd
+    kernel = make_2dgaussian_kernel(fwhm, size=size)  # sums to 1
+    return kernel.array
+
+
+def make_det_image(library, kernel_fwhm):
+    """
+    Make a detection image from a library of models.
+
+    The detection image is the weighted sum of the input models, where
+    the weights are SED weight divided by the read noise variance.
+
+    Parameters
+    ----------
+    library : `romancal.datamodels.ModelLibrary`
+        The input library of models.
+
+    kernel_fwhm : float
+        The full-width at half-maximum (FWHM) in pixels of the 2D
+        Gaussian kernel used to smooth the detection image.
+
+    Returns
+    -------
+    detection_data : 2D `numpy.ndarray`
+        The detection image data.
+
+    detection_error : 2D `numpy.ndarray`
+        The detection image (standard deviation) error.
+    """
+    if not isinstance(library, ModelLibrary):
+        raise TypeError("library input must be a ModelLibrary object")
+
+    # TODO: extend to different kernel shapes beyond Gaussian?
+    # would require different/additional kernel parameters
+    kernel = make_gaussian_kernel(kernel_fwhm)
+
+    log.info(f"Making detection image with kernel FWHM={kernel_fwhm}")
+
+    with library:
+        detection_data = 0.0
+        detection_var = 0.0
+        wht_sum = 0.0
+        for i, model in enumerate(library):
+            # TODO: SED weights to be defined in the asn file for each
+            # input filter image
+            try:
+                sed_weight = library.asn["products"][0]["members"][i]["sed_weight"]
```

Comment:
This does a single SED for right now, which makes sense, though we probably want to extend that.